File: library/src/main/java/com/pengrad/telegrambot/model/Gift.java
package com.pengrad.telegrambot.model;

import java.io.Serializable;
import java.util.Objects;

public class Gift implements Serializable {
    private final static long serialVersionUID = 0L;

    private String id;
    private Sticker sticker;
    private Integer star_count;
    private Integer total_count;
    private Integer remaining_count;

    public String id() {
        return id;
    }

    public Sticker sticker() {
        return sticker;
    }

    public Integer starCount() {
        return star_count;
    }

    public Integer totalCount() {
        return total_count;
    }

    public Integer remainingCount() {
        return remaining_count;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Gift gift = (Gift) o;
        return Objects.equals(id, gift.id) && Objects.equals(sticker, gift.sticker) && Objects.equals(star_count, gift.star_count) && Objects.equals(total_count, gift.total_count) && Objects.equals(remaining_count, gift.remaining_count);
    }

    @Override
    public int hashCode() {
        int result = Objects.hashCode(id);
        result = 31 * result + Objects.hashCode(sticker);
        result = 31 * result + Objects.hashCode(star_count);
        result = 31 * result + Objects.hashCode(total_count);
        result = 31 * result + Objects.hashCode(remaining_count);
        return result;
    }

    @Override
    public String toString() {
        return "Gift{" +
            "id='" + id + '\'' +
            ", sticker=" + sticker +
            ", star_count=" + star_count +
            ", total_count=" + total_count +
            ", remaining_count=" + remaining_count +
            '}';
    }
}

File: library/src/main/java/com/pengrad/telegrambot/model/gift/Gift.kt
package com.pengrad.telegrambot.model.gift

import com.pengrad.telegrambot.model.Chat
import com.pengrad.telegrambot.model.Sticker

data class Gift(
    @get:JvmName("id") val id: String,
    @get:JvmName("sticker") val sticker: Sticker,
    @get:JvmName("starCount") val starCount: Int,
    @get:JvmName("upgradeStarCount") val upgradeStarCount: Int? = null,
    @get:JvmName("totalCount") val totalCount: Int? = null,
    @get:JvmName("remainingCount") val remainingCount: Int? = null,
    @get:JvmName("personalTotalCount") val personalTotalCount: Int? = null,
    @get:JvmName("personalRemainingCount") val personalRemainingCount: Int? = null,
    @get:JvmName("isPremium") val isPremium: Boolean? = null,
    @get:JvmName("hasColors") val hasColors: Boolean? = null,
    @get:JvmName("publisherChat") val publisherChat: Chat? = null
)
